optuna/importance/_base.py

```diff
@@ -63,43 +63,55 @@ def evaluate(
         raise NotImplementedError
 
 
-def _get_distributions(study: Study, params: Optional[List[str]]) -> Dict[str, BaseDistribution]:
+def _get_distributions(
+    study: "optuna.Study", params: Optional[List[str]]
+) -> Dict[str, BaseDistribution]:
     _check_evaluate_args(study, params)
-
-    if params is None:
-        return intersection_search_space(study, ordered_dict=True)
-
-    # New temporary required to pass mypy. Seems like a bug.
-    params_not_none = params
-    assert params_not_none is not None
-
-    # Compute the search space based on the subset of trials containing all parameters.
     distributions = None
     for trial in study.trials:
         if trial.state != TrialState.COMPLETE:
             continue
 
-        trial_distributions = trial.distributions
-        if not all(name in trial_distributions for name in params_not_none):
-            continue
+        if params is not None:
+            trial_distributions = trial.distributions
+            if not all(name in trial_distributions for name in params):
+                continue
 
         if distributions is None:
-            distributions = dict(
-                filter(
-                    lambda name_and_distribution: name_and_distribution[0] in params_not_none,
-                    trial_distributions.items(),
-                )
-            )
+            distributions = {}
+            for param_name, param_distribution in trial.distributions.items():
+                if (params is None) or (params is not None and param_name in params):
+                    distributions[param_name] = param_distribution
             continue
 
-        if any(
-            trial_distributions[name] != distribution
-            for name, distribution in distributions.items()
-        ):
-            raise ValueError(
-                "Parameters importances cannot be assessed with dynamic search spaces if "
-                "parameters are specified. Specified parameters: {}.".format(params)
-            )
+        delete_list = []
+        for param_name, param_distribution in distributions.items():
+            if param_name not in trial.distributions:
+                delete_list.append(param_name)
+            else:
+                trial_param_distribution = trial.distributions[param_name]
+                if isinstance(param_distribution, CategoricalDistribution):
+                    if trial_param_distribution != param_distribution:
```

Comment:
Can we assert this instead since `_get_distributions` is a private method and only used internally?